src/core/experimental/array.d

```diff
@@ -0,0 +1,2299 @@
+     *            input range of such elements, or another `Array`
+     *
+     * Returns:
+     *      a reference to this array
+     *
+     * Complexity: $(BIGOH n + m), where `m` is the number of elements in `rhs`.
+     */
+    auto ref opOpAssign(string op, U)(auto ref U rhs)
+        if (op == "~" &&
+            (is (U == typeof(this))
+             || is (U : T)
+             || (is (U == V[], V) && is(V : T))
+            ))
+    {
+        insert(length, rhs);
+        return this;
+    }
+
+    ///
+    static if (is(T == int))
+    @safe unittest
+    {
+        Array!int a;
+        auto a2 = Array!int(4, 5);
+        assert(a.empty);
+
+        a ~= 1;
+        a ~= [2, 3];
+        assert(equal(a, [1, 2, 3]));
+
+        // append an input range
+        a ~= a2;
+        assert(equal(a, [1, 2, 3, 4, 5]));
+        a2.front = 0;
+        assert(equal(a, [1, 2, 3, 4, 5]));
+    }
+
+    ///
+    bool opEquals()(auto ref typeof(this) rhs) const
+    {
+        return payload.equal(rhs);
+    }
+
+    ///
+    static if (is(T == int))
+    @safe unittest
+    {
+        auto arr1 = Array!int(1, 2);
+        auto arr2 = Array!int(1, 2);
+        auto arr3 = Array!int(2, 3);
+        assert(arr1 == arr2);
+        assert(arr2 == arr1);
+        assert(arr1 != arr3);
+        assert(arr3 != arr1);
+        assert(arr2 != arr3);
+        assert(arr3 != arr2);
+    }
+
+    ///
+    int opCmp(U)(auto ref U rhs)
+    if ((is(U == Array!V, V) || is(U == V[], V)) && is(V : T))
+    {
+        auto r1 = this;
+        auto r2 = rhs;
+        while (r1.length && r2.length)
+        {
+            if (r1[0] < r2[0])
+                return -1;
+            else if (r1[0] > r2[0])
+                return 1;
+            r1 = r1[1 .. $];
+            r2 = r2[1 .. $];
+        }
+        // arrays are equal until here, but it could be that one of them is shorter
+        if (!r1.length && !r2.length)
+            return 0;
+        return (r1.length == 0) ? -1 : 1;
+    }
+
+    ///
+    static if (is(T == int))
+    @safe unittest
+    {
+        auto arr1 = Array!int(1, 2);
+        auto arr2 = Array!int(1, 2);
+        auto arr3 = Array!int(2, 3);
+        auto arr4 = Array!int(0, 3);
+        assert(arr1 <= arr2);
+        assert(arr2 >= arr1);
+        assert(arr1 < arr3);
+        assert(arr3 > arr1);
+        assert(arr4 < arr1);
+        assert(arr4 < arr3);
+        assert(arr3 > arr4);
+    }
+
+    static if (is(T == int))
+    @safe unittest
+    {
+        auto arr1 = Array!int(1, 2);
+        auto arr2 = [1, 2];
+        auto arr3 = Array!int(2, 3);
+        auto arr4 = [0, 3];
+        assert(arr1 <= arr2);
+        assert(arr2 >= arr1);
+        assert(arr1 < arr3);
+        assert(arr3 > arr1);
+        assert(arr4 < arr1);
+        assert(arr4 < arr3);
+        assert(arr3 > arr4);
+    }
+
+    ///
+    auto toHash()
+    {
+        // will be safe with 2.082
```

Comment:
It's 2.082 now.

Reply:
Good catch. Will remove obsolete comment.

Reply:
Well, it was my comment after all ;-)
You should add a unittest to ensure that it's actually @safe though.